### PR TITLE
sync timezone every 10 min and change the cron timezone

### PR DESCRIPTION
**Pull request by ChanYiLin**
Signed-off-by: Jack Lin <chanyi.jack.lin@infuseai.io>

** PR checklist **

**What type of PR is this?**
feature

**What this PR does / why we need it**:
to sync timezone every 10 min and change the cron timezone

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

### CODE REVIEW

File: controllers/phschedule_controller.go [hunk 1]
```diff
@@ -49,9 +49,12 @@ type PhScheduleReconciler struct {
 	GraphqlClient     *graphql.GraphqlClient
 }
 
-func (r *PhScheduleReconciler) buildPhJob(phSchedule *primehubv1alpha1.PhSchedule) (*primehubv1alpha1.PhJob, error) {
+func (r *PhScheduleReconciler) buildPhJob(phSchedule *primehubv1alpha1.PhSchedule, location string) (*primehubv1alpha1.PhJob, error) {
 	log := r.Log.WithValues("phschedule", phSchedule.Name)
-	t := time.Now().UTC()
+
+	timeLocation, err := time.LoadLocation(location) // generate job name with timestamp based on current timezone
```
[the remark this answers is not in the document]
Comment (ChanYiLin) [reply controllers/phschedule_controller.go]: fixed

File: controllers/phschedule_controller.go [hunk 2]
```diff
@@ -116,6 +119,16 @@ func (r *PhScheduleReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error)
 		log.Info("Finished Reconciling phSchedule ", "phSchedule", phSchedule, "ReconcileTime", time.Since(startTime))
 	}()
 
+	// fetch timezone from system to sync the timezone
```
[the remark this answers is not in the document]
Comment (ChanYiLin) [reply controllers/phschedule_controller.go]: moved

File: controllers/phschedule_controller.go [hunk 3]
```diff
@@ -242,7 +254,7 @@ func (r *PhScheduleReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error)
 		phScheduleCron.c.Remove(phScheduleCron.entryID)
 	}
 
-	entryID, _ = phScheduleCron.c.AddFunc(recurrence, createPhJob)
+	entryID, _ = phScheduleCron.c.AddFunc("CRON_TZ="+location+" "+recurrence, createPhJob) // use system timezone
```
Comment (popcornylu) [inline controllers/phschedule_controller.go]: Should check the error returned from `AddFunc`

File: controllers/phschedule_controller.go [hunk 4]
```diff
@@ -242,7 +254,18 @@ func (r *PhScheduleReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error)
 		phScheduleCron.c.Remove(phScheduleCron.entryID)
 	}
 
-	entryID, _ = phScheduleCron.c.AddFunc(recurrence, createPhJob)
+	entryID, err = phScheduleCron.c.AddFunc("CRON_TZ="+location+" "+recurrence, createPhJob) // use system timezone
+	if err != nil {
+		log.Error(err, "controller cannot create cron job")
+		phSchedule.Status.Invalid = false
```
Comment (popcornylu) [inline controllers/phschedule_controller.go]: Should be `Invalid = true`?

**Comment by popcornylu** [comment]
LGTM

**Pull request merged**